pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -197,6 +210,20 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
+type ResourceSelector struct {
+	// name of an object within a claimed group/resource.
+	// It matches the metadata.name field of the underlying object.
+	//
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// namespace containing the named object. Matches metadata.namespace field.
+	// If "name" is unset, all objects from the namespace are being claimed.
+	//
+	// +required
```

Comment:
Is this required if we use a label selector? Or are you intending for `resourceSelector` and label selection to be mutually exclusive?

Reply:
This won't be required if we use a label selector. For now, the constrained declared here is ok but once we add label selection into this struct, I guess we'll need CEL to express mutual exclusivity.

Reply:
Kube has (or had, not up-to-date on the migration) a solution for this on the `VolumeSource` of a PV. [See this comment](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L48).

I'll dig around some more if I can see the exact mechanism, I don't think we need CEL for this unless it's the new way of doing things.

Reply:
I have precedence of mutual exlusitivity using CEL here: https://github.com/kcp-dev/kcp/pull/2222/files#diff-228a80bf14ca2df5c0d6053a9a4ed5f5ddb5d74a13c2618584741e2723afbacbR190
```
// +kubebuilder:validation:XValidation:rule="self.all != has(self.resourceSelector)",message="either \"all\" or \"resourceSelector\" must be set"
type PermissionClaim struct {
```

Reply:
Awesome. I didn't see the `XValidation` marker when looking at kubebuilder markers, but that's perfect.